paddle/framework/op_registry.h

```diff
@@ -44,52 +45,45 @@ class OpProtoAndCheckerMaker {
 
  protected:
   struct VariableBuilder {
-    VarProto* var_;
-    std::function<void()> on_multiple_;
-    std::function<void()> on_temporary_;
+    OpProto::Var* var_;
 
     VariableBuilder& SetMultiple() {
-      var_->set_multiple(true);
-      on_multiple_();
+      var_->set_duplicable(true);
       return *this;
     }
 
     VariableBuilder& SetTemporary() {
-      PADDLE_ENFORCE(bool(on_temporary_), "Cannot set temporary");
-      var_->set_temporary(true);
-      on_temporary_();
+      var_->set_intermediate(true);
       return *this;
     }
 
     VariableBuilder& IgnoreGradient() {
```

Comment:
Rename to `NoGradient`.

However, I do not think `NoGradient` is a straight name. Maybe we should change it to another name in following PR @Canpio @qingqing01